### PR TITLE
Introduce new ZoneGetter and refactor related packages and tests

### PR DESCRIPTION
**Pull request by songrx1997**
/assign @swetharepakula

### CODE REVIEW

File: pkg/utils/zonegetter/fake.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
+/*
```
Comment (swetharepakula) [inline pkg/utils/zonegetter/fake.go:1]: can you move this into its own commit?

File: pkg/utils/zonegetter/fake.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+				},
+			},
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InitializeFakeZoneGetter(zoneGetter *ZoneGetter) {
```
Comment (swetharepakula) [inline pkg/utils/zonegetter/fake.go:80]: I wonder instead of hardcoding this, we should have a utils function that returns a list of nodes and this would sync the node informer in the fake to that?

Or perhaps we should make it so that this uses the same node informer that already exists, so it should mean as long as the node informer is maintained in the test, we don't need to do anything for the fake zone getter.

Wdyt?

File: pkg/utils/zonegetter/fake.go (hunk 1 above)
Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go:1]: Done

File: pkg/utils/zonegetter/fake.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	informerv1 "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/ingress-gce/pkg/utils"
+)
+
+// NewFakeZoneGetter initialize a fake ZoneGetter in normal mode.
+func NewFakeZoneGetter() *ZoneGetter {
```
[the remark this answers is not in the document]
Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go]: Removed this since the `NewZoneGetter` did this and can be used for tests with a fake nodeInformer.
Added another function `FakeNodeInformer` to create a fake nodeInformer for tests.

File: pkg/utils/zonegetter/fake.go (hunk 2 above)
Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go]: Change this to take a nodeInformer and add nodes to it.

File: pkg/utils/zonegetter/fake.go [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+		if nodeZone == zone {
+			if err := nodeIndexer.Delete(node); err != nil {
+				t.Errorf("Failed to delete node %q in zone %q, err - %v", node.Name, zone, err)
+			}
+		}
+	}
+}
+
+// Fake add nodes adds fake nodes to the ZoneGetter in the provided zone.
+func FakeAddNodes(zoneGetter *ZoneGetter, newZone string, instances ...string) error {
```
Comment (swetharepakula) [inline pkg/utils/zonegetter/fake.go:55]: maybe AddFakeNodes

File: pkg/neg/manager_test.go [hunk 5]
```diff
@@ -1417,6 +1421,11 @@ func TestSyncNodesConditions(t *testing.T) {
 		addZones      map[string][]string
 		deleteZones   []string
 	}{
+		{
+			desc:       "vm ip port neg, zones are the same",
+			expectSync: false,
+			negType:    negtypes.VmIpPortEndpointType,
+		},
```
[the remark this answers is not in the document]
Comment (songrx1997) [reply pkg/neg/manager_test.go]: Maybe this got moved when debugging the tests, reverted the move.

File: pkg/utils/zonegetter/fake.go (hunk 4 above)
Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go]: Done

File: pkg/utils/zonegetter/fake.go [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+				},
+			},
+		}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func FakeInitializeNodeInformer(nodeInformer cache.SharedIndexInformer) {
```
[the remark this answers is not in the document]
Comment (songrx1997) [reply pkg/utils/zonegetter/fake.go]: Changed name a added the comment. The nodes added to the informer is only for NEG tests. For the tests in other packages, only fake informer is required.
